File: 프로그래머스/2/181188. 요격 시스템/요격 시스템.java
import java.util.*;
class Solution {
    public int solution(int[][] targets) {
        int answer = 0;

        Arrays.sort(targets, (o1,o2)->o1[1]-o2[1]);

        answer += 1;
        double last = targets[0][1] -0.5;
        for(int i = 1; i< targets.length; i++){
            if(targets[i][0] <= last){
                // System.out.println("skip " + targets[i][0] + " " + targets[i][1]);
                continue;
            } else {
                answer += 1;
                last = (targets[i][1] - 0.5);
                // System.out.println("add " + targets[i][0] + " " + targets[i][1]);
            }
        }

        return answer;
    }
}

File: 프로그래머스/2/181188. 요격 시스템/요격 시스템.kt
class Solution {
    fun solution(targets: Array<IntArray>): Int {
        var answer: Int = 0

        var e = -1
        var s = -1
        targets.sortedWith( compareBy<IntArray> { it[1]}.thenBy { it[0] })
            .forEach{
                if(it[0] >= e){
                    answer += 1
                    s = it[0]
                    e = it[1]
                }
            }
        return answer
    }
}